src/download/src/lib.rs

```diff
@@ -612,7 +615,12 @@ pub mod hyper_base {
             _ => return Err(format!("unsupported URL scheme: '{}'", url.scheme()).into())
         };
 
-        let mut res = try!(client.get(url.clone()).send()
+        use std::cmp;
+        let byte = cmp::max(resume_from, 0);
+        let header = Range::Bytes(vec![ByteRangeSpec::AllFrom(byte)]);
+        let req = client.get(url.clone()).header(header);
+
+        let mut res = try!(req.send()
                            .chain_err(|| "failed to make network request"));
         if res.status != self::hyper::Ok {
```

Comment:
Writing the tests will be tricky. It'd require starting up an HTTP server and serving partial contents of a file. The currently depended on version of hyper doesn't allow servers to be torn down when the test is done. It might need to wait for an upgrade to that...